server/users/users.controller.ts

```diff
@@ -49,11 +60,51 @@ class UsersController {
 	}
 
 	public deleteUser = async (req: RequestWithUser, res: Response, next: NextFunction) => {
+		const userId: number = Number(req.user.id);
+		const userData: RemoveUserCommand = req.body;
+		var user : UserModel;
+
+		try {
+			user = await this.authService.authenticate({ email: req.user.email, password: userData.password })
+
+		} catch {
+			const userData = await this.userService.getById(userId);
```

Comment:
Can we just do `res.redirect("/users/id")` here?

Same in the try block below, we probably don't need `returnUrlQueryParam` and we can just redirect to `/`

Reply:
Sure, but in that case can I return the error message `"Incorrect password."`?

Reply:
Hm yeah, let's leave it then. Sometime in the future we could send the error messages in some more generic way, so we'd be able to do something like `res.redirect("/users/id?err=Incorrect%20password")`. I'll write it down somewhere.